source/common/upstream/edf_scheduler.h

```diff
@@ -64,7 +66,92 @@ template <class C> class EdfScheduler : public Scheduler<C> {
 
   bool empty() const override { return queue_.empty(); }
 
+  // Creates an EdfScheduler with the given weights and their corresponding
+  // entries, and emulating a number of initial picks to be performed. Note that
+  // the internal state of the scheduler will be very similar to creating an empty
+  // scheduler, adding the entries one after the other, and then performing
+  // "picks" pickAndAdd operation without modifying the entries' weights.
+  // The only thing that may be different is that entries with the same weight
+  // may be chosen a bit differently (the order_offset_ values may be different).
+  // Breaking the ties of same weight entries will be kept in future picks from
+  // the scheduler.
+  static EdfScheduler<C> createWithPicks(const std::vector<std::shared_ptr<C>>& entries,
+                                         std::function<double(const C&)> calculate_weight,
+                                         uint32_t picks) {
+    EDF_TRACE("Creating an EDF-scheduler with {} weights and {} pre-picks.", entries.size(), picks);
+    // Assume no non-positive weights.
+    ASSERT(std::none_of(entries.cbegin(), entries.cend(),
+                        [&calculate_weight](const std::shared_ptr<C>& entry) {
+                          return calculate_weight(*entry) <= 0;
+                        }));
+
+    // Nothing to do if there are no entries.
+    if (entries.size() == 0) {
+      return EdfScheduler<C>();
+    }
+
+    // Augment the weight computation to add some epsilon to each entry's
+    // weight to avoid cases where weights are multiplications one of each other.
+    auto aug_calculate_weight = [&calculate_weight](const C& entry) -> double {
+      return calculate_weight(entry) + 0.00000000001;
```

Comment:
nit: Consider `1e-10`

Reply:
Good idea, makes it more readable. Done.